widget/ResourceGateway.swift

```diff
@@ -0,0 +1,50 @@
+//  Copyright © 2020 Onegini. All rights reserved.
+
+import UIKit
+import OneginiSDKiOS
+
+class ResourceGateway {
+    
+    func fetchImplicitResources(profile: ONGUserProfile, completion: @escaping (String?) -> Void) {
+        if isProfileImplicitlyAuthenticated(profile) {
```

Comment:
At the beginning, the widget works fine, however after some time, I started to receive 401 for each widget refresh. It seems that implicit access token gets revoked on TS (timeout?) and we still detect that profile as implicitly authenticated and try with the same (revoked) implicit access token. So actually two things to consider (imho):
- from what I checked, there is not implicit logout method or something similar so once implicitly authenticated, the implicit access token never expires or is cleared on SDK side, only way is to call `authenticateUserImplicitly` method again
- if we remove the profile check and authenticate implicitly each time, it works every time for me so maybe we could at least change that